pkg/task/registration.go

```diff
@@ -0,0 +1,81 @@
+package task
+
+import (
+	"sync"
+
+	"github.com/Peripli/service-broker-proxy/pkg/platform"
+	"github.com/Peripli/service-broker-proxy/pkg/sm"
+	"github.com/sirupsen/logrus"
+)
+
+type SBProxyRegistration struct {
+	group          *sync.WaitGroup
+	platformClient platform.Client
+	smClient       sm.Client
+	smHost         string
+}
+
+func New(group *sync.WaitGroup, platoformClient platform.Client, smClient sm.Client, smHost string) *SBProxyRegistration {
+	return &SBProxyRegistration{
+		group:          group,
+		platformClient: platoformClient,
+		smClient:       smClient,
+		smHost:         smHost,
+	}
+}
+
+func (r SBProxyRegistration) Run() {
+	r.group.Add(1)
+	defer r.group.Done()
+	r.run()
+}
+
+func (r SBProxyRegistration) run() {
+	logrus.Debug("Running broker registration task...")
+	registeredBrokers, err := r.platformClient.GetBrokers()
+	if err != nil {
+		logrus.Error("An error occurred while obtaining already registered brokers: ", err)
+		return
+	}
+
+	proxyBrokers, err := r.smClient.GetBrokers()
+	if err != nil {
+		logrus.Error("An error occurred while obtaining brokers that have to be registered at the platform: ", err)
+		return
+	}
+
+	updateBrokerRegistrations(r.createBrokerRegistration, proxyBrokers, registeredBrokers)
+	updateBrokerRegistrations(r.deleteBrokerRegistration, registeredBrokers, proxyBrokers)
+}
+
+func (r SBProxyRegistration) deleteBrokerRegistration(broker *platform.ServiceBroker) {
+	deleteRequest := &platform.DeleteServiceBrokerRequest{
+		Guid: broker.Guid,
+	}
+	if err := r.platformClient.DeleteBroker(deleteRequest); err != nil {
+		logrus.Error("Error during broker deletion: ", err)
+	}
+}
+
+func (r SBProxyRegistration) createBrokerRegistration(broker *platform.ServiceBroker) {
+	createRequest := &platform.CreateServiceBrokerRequest{
+		Name:      "sm-proxy-" + broker.Name,
+		BrokerURL: r.smHost + "/" + broker.Guid,
+		SpaceGUID: broker.Guid,
+	}
+	if _, err := r.platformClient.CreateBroker(createRequest); err != nil {
+		logrus.Error("Error during broker registration: ", err)
+	}
+}
+
+func updateBrokerRegistrations(updateOp func(broker *platform.ServiceBroker), a, b []platform.ServiceBroker) {
```

Comment:
there is a bug here - currently guid of the sm broker is compared to the guild of the proxy-broker registration at the platform - these will obviously always be different sicne we do not control what the id at the platform is.